.github/workflows/branch.yml

```diff
@@ -0,0 +1,46 @@
+name: nf-core branch protection
+# This workflow is triggered on PRs to `main`/`master` branch on the repository
+# It fails when someone tries to make a PR against the nf-core `main`/`master` branch instead of `dev`
+on:
+  pull_request_target:
+    branches:
+      - main
+      - master
```

Comment:
Would honestly prefer if master was the only branch here. Seems strange to use main.